proposals/0110-exponential-fee-for-write-lock-accounts.md

```diff
@@ -0,0 +1,138 @@
+extends: (optional - fill this in if the SIMD extends the design of a previous
+ SIMD)
+---
+
+## Summary
+
+In a permissionless environment with low fees, users often submit transactions
+for opportunistic trades without considering the success probability. Almost
+half of the Compute Units (CUs) in a block are allocated to failed transactions,
+leading to undesirable scenario where large portion of compute powers primarily
+serving failed DeFi arbitrage transactions. To address this, the proposed
+feature introduces economic back pressure, discouraging spam activities and
+ensuring efficient network resource utilization. It suggests tracking the
+Exponential Moving Average (EMA) of contentious accounts' CU utilization
+per block and exponentially increasing the cost to write-lock these accounts
+if their utilization remains high.
+
+## Motivation
+
+The motivation behind this feature is to introduce economic back pressure,
+dissuading DeFi spammers from overwhelming the network. DeFi spam, defined as
+opportunistic trades with a positive return on investment, currently occupies
+almost half of the CUs in a block as failed transactions. Economic back
+pressure aims to create a deterrent for such spam activities, ensuring network
+resources are efficiently utilized and preventing continuous block congestion
+caused by failed DeFi spam transactions.
+
+## Alternatives Considered
+
+While the priority fee serves to mitigate low-cost spams by decreasing the
+likelihood of less prioritized transactions being included, it cannot entirely
+eliminate the inclusion of spam transactions in a block. As long as there
+remains a chance, no matter how small, to inexpensively include transactions,
+the incentive for spamming persists. The proposed feature recognizes that the
+current mechanisms have limitations in fully addressing the spam issue, and
+thus, seeks to introduce a more robust solution to discourage opportunistic
+trades and ensure a more secure and efficient network environment.
+
+## New Terminology
+
+- *compute-unit utilization*: denominated in `cu`, it represents total
+ compute-units applied to a given resource.
+- *cost rate*: denominated in `lamport/cu`, it represents the cost per
+compute-unit at a given condition.
+- *compute unit pricer*: a componenet tracks Exponential Moving Average of
+*compute-unit utilization*, applies a pricing algorithm to provide current
+*cost rate*.
+- *write lock fee*: denominated in `lamport`, it is fee dedicated for write
+lock an account, calculated as `compute-unit-pricer.cost-rate() * transaction.requested_cu()`.
+
+## Design Highlights
+
+- Account Association with Compute Unit Pricer:
+  - Accounts are associated with a *compute unit pricer*, and the *runtime*
+  maintains an LRU cache of actively contentious accounts' public keys and
+  their *compute unit pricers*.
+  - Alternatively, each account can have its *compute unit pricer* stored
+  onchain, which would require modifying accounts.
```

Comment:
....but if there are spare bytes, this can be done cheaply (and you can avoid the cache and have a price on literally every account) with two u64s: store `last_slot_touched` and an int representing `log(fee rate) / log(1.01)`

Reply:
That would require storage forever in the state. An LRU cache should be sufficient.

Reply:
only if it is big enough!

Reply:
I think it only needs to be 2*worst case evictions per block